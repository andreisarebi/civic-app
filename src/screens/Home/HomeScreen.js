import React from 'react';
import { StyleSheet, ScrollView, View, Text } from 'react-native';
import Banner, { mapAlertLevelToColor } from './Banner';
import Colors from '../../styles/colors';
import DailyTasks from './DailyTasks';
import UpcomingActivism from './UpcomingActivism';

// eventually this will be in state
const alerts = [
  {
    title: 'Not Registered to Vote?',
    subtitle: 'Click here to get started',
    level: 'high',
  },
  {
    title: 'Something is going on in your city!',
    subtitle: 'Click here to find out more',
    level: 'medium',
  },
];

const dailyTasks = [
  {
    id: 'task1',
    heading: 'Narrow your matches',
    content: 'Do you think the U.S. should be more lenient on immigration?',
    userAgrees: null
  },
  {
    id: 'task2',
    heading: 'Narrow your matches',
    content: 'Should there be limits on donations to political campaigns?',
    userResponse: ''
  },
]

const upcomingActivism = [
  {
    id: 'activism1',
    title: `Women's March 2018`,
    imgUrl: '',
    dateTime: new Date(2018, 11, 18, 12)
  },
  {
    id: 'activism2',
    title: 'Some Other March 2018',
    imgUrl: '',
    dateTime: new Date(2018, 11, 19, 12)
  },
]

class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
  };

  render() {
    return (
      <ScrollView
        contentContainerStyle={styles.container}
      >
        {alerts.map((alert, index) => (
          <Banner
            key={index}
            color={mapAlertLevelToColor(alert.level)}
            icon="megaphone"
            style={styles.banner}
            title={alert.title}
            subtitle={alert.subtitle}
          />
        ))}

        {/* Daily Tasks */}
        <Text style={styles.sectionHeader}>DAILY TASKS</Text>
        <DailyTasks data={dailyTasks}/>

        {/* Upcoming Activism */}
        <Text style={styles.sectionHeader}>UPCOMING ACTIVISM</Text>
        <UpcomingActivism data={upcomingActivism}/>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 8,
    paddingTop: 0,
  },
  banner: {
    marginTop: 8,
  },
  sectionHeader: {
    fontSize: 14,
    margin: 18,
    color: Colors.gray
  }
});

export default HomeScreen;
