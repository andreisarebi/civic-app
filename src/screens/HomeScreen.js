import React from 'react';
import { View, Text } from 'react-native';
import styles from './styles';

class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
  };

  render() {
    return (
      <View style={styles.container}>
        <Text>This is the Home Screen</Text>
      </View>
    );
  }
}

export default HomeScreen;
