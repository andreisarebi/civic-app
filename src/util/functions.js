import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
