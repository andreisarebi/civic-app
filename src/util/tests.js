import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
