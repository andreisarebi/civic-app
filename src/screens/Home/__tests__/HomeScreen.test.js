import { createSnapshot } from '../../../util/tests';
import HomeScreen from '../HomeScreen';

describe('HomeScreen', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(HomeScreen);
    expect(snapshot).toMatchSnapshot();
  });
});
