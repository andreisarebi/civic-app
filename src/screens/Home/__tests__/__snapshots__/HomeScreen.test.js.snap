// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingTop": 0,
    }
  }
>
  <Banner
    color="#eb5757"
    icon="megaphone"
    style={
      Object {
        "marginTop": 8,
      }
    }
    subtitle="Click here to get started"
    title="Not Registered to Vote?"
  />
  <Banner
    color="#126390"
    icon="megaphone"
    style={
      Object {
        "marginTop": 8,
      }
    }
    subtitle="Click here to find out more"
    title="Something is going on in your city!"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#9B9B9B",
        "fontSize": 14,
        "margin": 18,
      }
    }
  >
    DAILY TASKS
  </Text>
  <DailyTasks
    data={
      Array [
        Object {
          "content": "Do you think the U.S. should be more lenient on immigration?",
          "heading": "Narrow your matches",
          "id": "task1",
          "region": "California",
          "userResponse": "",
        },
        Object {
          "content": "Should there be limits on donations to political campaigns?",
          "heading": "Narrow your matches",
          "id": "task2",
          "region": "California",
          "userResponse": "",
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#9B9B9B",
        "fontSize": 14,
        "margin": 18,
      }
    }
  >
    UPCOMING ACTIVISM
  </Text>
  <UpcomingActivism
    data={
      Array [
        Object {
          "eventDate": 2018-12-18T17:00:00.000Z,
          "id": "activism1",
          "imgUrl": "",
          "title": "Women's March 2018",
        },
        Object {
          "eventDate": 2018-12-19T17:00:00.000Z,
          "id": "activism2",
          "imgUrl": "",
          "title": "Some Other March 2018",
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#9B9B9B",
        "fontSize": 14,
        "margin": 18,
      }
    }
  >
    IN THE NEWS
  </Text>
  <NewsCard
    createdAt={2018-08-18T16:00:00.000Z}
    img={1}
    title="Does Gavin Newsom represent a shift in California Democratic Party?"
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "margin": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#555",
          "fontSize": 16,
        }
      }
    >
      You're all caught up for today.
    </Text>
    <Icon
      color="#DADADA"
      containerStyle={
        Object {
          "margin": 20,
        }
      }
      name="megaphone"
      raised={false}
      reverse={false}
      reverseColor="white"
      size={50}
      type="entypo"
      underlayColor="white"
    />
  </Component>
</ScrollView>
`;
